### PR TITLE
Reward distribution features test suite

### PR DESCRIPTION
**Pull request by irina-pereiaslavskaia**
New tests for Reward Distribution Features

### CODE REVIEW

**Review by alexeykiselev — DISMISSED** [review]

File: itests/testdata/reward_distribution_features_testdata.go
Comment (nickeskov) [inline itests/testdata/reward_distribution_features_testdata.go:172]: HTTP request count will be decreased if you create a new variable with result of `utl.GetCurrentReward(suite, height)` function call and reuse it (fix it in all places).

File: itests/utilities/common_tx_utility.go
Comment (nickeskov) [inline itests/utilities/common_tx_utility.go:484]: Add `default` case

File: itests/utilities/common_tx_utility.go
Comment (nickeskov) [inline itests/utilities/common_tx_utility.go:505]: `suite.FailNowf` can be used here instead of `require` below.

File: itests/utilities/common_tx_utility.go
Comment (nickeskov) [inline itests/utilities/common_tx_utility.go:726]: The same issue with `suite.FailNowf`

File: itests/utilities/common_tx_utility.go
Comment (nickeskov) [inline itests/utilities/common_tx_utility.go:777]: The variable is initialized nowhere.

File: itests/utilities/common_tx_utility.go
Comment (nickeskov) [inline itests/utilities/common_tx_utility.go:783]: Else?

File: itests/utilities/common_tx_utility.go
Comment (nickeskov) [inline itests/utilities/common_tx_utility.go:516]: The same issue with `suite.FailNowf`

File: itests/utilities/common_tx_utility.go
Comment (nickeskov) [inline itests/utilities/common_tx_utility.go:485]: Please, create constants for these strings and use them

File: itests/utilities/reward/reward.go
Comment (nickeskov) [inline itests/utilities/reward/reward.go:73]: Is it intended log record? I mean, it's unusual that it comes without additional info

File: itests/utilities/reward/reward.go
Comment (nickeskov) [inline itests/utilities/reward/reward.go:78]: The same as above

File: itests/testdata/reward_distribution_rolllback_testdata.go
Comment (nickeskov) [inline itests/testdata/reward_distribution_rolllback_testdata.go:27]: HTTP request count will be decreased if you create a new variable with result of `utl.GetCurrentReward(suite, height)` function call and reuse it (fix it in all places).

**Review by nickeskov — CHANGES_REQUESTED** [review]

File: itests/config/genesis_settings.go
Comment (irina-pereiaslavskaia) [reply itests/config/genesis_settings.go:137]: I left the Decoder as it is, but add f.Close()

File: itests/config/genesis_settings.go
Comment (irina-pereiaslavskaia) [reply itests/config/genesis_settings.go:248]: Fix.

File: itests/fixtures/reward_preactivated_features_fixtures.go
Comment (irina-pereiaslavskaia) [reply itests/fixtures/reward_preactivated_features_fixtures.go:24]: Done

File: itests/fixtures/reward_preactivated_features_fixtures.go
Comment (irina-pereiaslavskaia) [reply itests/fixtures/reward_preactivated_features_fixtures.go:25]: Done

File: itests/node_client/grpc_client.go
Comment (irina-pereiaslavskaia) [reply itests/node_client/grpc_client.go:36]: Fix

File: itests/reward_distribution_info_internal_test.go
Comment (irina-pereiaslavskaia) [reply itests/reward_distribution_info_internal_test.go:37]: Done

File: itests/reward_distribution_info_internal_test.go
Comment (irina-pereiaslavskaia) [reply itests/reward_distribution_info_internal_test.go:41]: Done

File: itests/reward_distribution_info_internal_test.go
Comment (irina-pereiaslavskaia) [reply itests/reward_distribution_info_internal_test.go:56]: Done

File: itests/reward_distribution_info_internal_test.go
Comment (irina-pereiaslavskaia) [reply itests/reward_distribution_info_internal_test.go:80]: Done

File: itests/testdata/reward_distribution_features_testdata.go
Comment (irina-pereiaslavskaia) [reply itests/testdata/reward_distribution_features_testdata.go:15]: Done

File: itests/utilities/common_tx_utility.go
Comment (irina-pereiaslavskaia) [reply itests/utilities/common_tx_utility.go:485]: Done

File: itests/utilities/common_tx_utility.go
Comment (irina-pereiaslavskaia) [reply itests/utilities/common_tx_utility.go:484]: Done

File: itests/utilities/common_tx_utility.go
Comment (irina-pereiaslavskaia) [reply itests/utilities/common_tx_utility.go:505]: Done

File: itests/utilities/common_tx_utility.go
Comment (irina-pereiaslavskaia) [reply itests/utilities/common_tx_utility.go:516]: Fix

File: itests/utilities/common_tx_utility.go
Comment (irina-pereiaslavskaia) [reply itests/utilities/common_tx_utility.go:726]: Done

File: itests/utilities/common_tx_utility.go
Comment (irina-pereiaslavskaia) [reply itests/utilities/common_tx_utility.go:783]: Fix

File: itests/utilities/reward/reward.go
Comment (irina-pereiaslavskaia) [reply itests/utilities/reward/reward.go:73]: I think, yes, it is, because of debugging convenience.

File: itests/utilities/reward/reward.go
Comment (irina-pereiaslavskaia) [reply itests/utilities/reward/reward.go:78]: Yes.

File: itests/testdata/reward_distribution_rolllback_testdata.go
Comment (irina-pereiaslavskaia) [reply itests/testdata/reward_distribution_rolllback_testdata.go:27]: Done

File: itests/testdata/reward_distribution_features_testdata.go
Comment (irina-pereiaslavskaia) [reply itests/testdata/reward_distribution_features_testdata.go:172]: Done

File: itests/config/genesis_settings.go
Comment (irina-pereiaslavskaia) [reply itests/config/genesis_settings.go:194]: If len == 0 or greater than 1 default values will be used.

File: itests/utilities/common_tx_utility.go
Comment (nickeskov) [reply itests/utilities/common_tx_utility.go:487]: I'm not agree. This is user provided data. You should handle any possible cases.

File: itests/config/genesis_settings.go
Comment (irina-pereiaslavskaia) [reply itests/config/genesis_settings.go:122]: Done

File: itests/fixtures/base_fixtures.go
Comment (irina-pereiaslavskaia) [reply itests/fixtures/base_fixtures.go:17]: Done

File: itests/reward_distribution_info_internal_test.go
Comment (irina-pereiaslavskaia) [reply itests/reward_distribution_info_internal_test.go]: Fix for some tests.

File: itests/config/genesis_settings.go
Comment (nickeskov) [reply itests/config/genesis_settings.go]: Fixed in [this commit](https://github.com/wavesplatform/gowaves/pull/1151/commits/ad47b3880be3979ab4a8dafd12967662126af6d3)

**Review by nickeskov — DISMISSED** [review]

**Review by nickeskov — DISMISSED** [review]

**Review by nickeskov — APPROVED** [review]

**Comment by nickeskov** [comment]
@DrBlast waiting for your approval

**Review by DrBlast — APPROVED** [review]

**Pull request merged**
